docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -237,11 +328,21 @@ Edit **package.json** to use [EAS Build hooks](/build-reference/npm-hooks.mdx) t
 
 > Don't forget to add executable permissions to **eas-build-pre-install.sh** and **eas-build-on-success.sh**. Run `chmod +x eas-hooks/*.sh`.
 
+### 5.1. Patch **app/build.gradle**
+
+> This step will be redundant in the future.
+
+The Android build command that you use to produce the test build is `./gradlew :app:assembleRelease :app:assembleAndroidTest -DtestBuildType=release`. Notice that it consists of two Gradle tasks. Unfortunately, when building the `*AndroidTest` task, some versions of the `expo-modules-code` module change what native libraries are included in the app binary. Those settings don't work with settings for `assembleRelease`.
+
+To fix the problem, add the `pickFirsts` list under `android.packagingOptions` in your **android/app/build.gradle**. The `pickFirsts` property overrides the setting for your project.
+
+<DiffBlock source="/static/diffs/e2e-tests-pickfirsts.diff" />
```

Comment:
I have no idea what I described here. I could have used the wrong words. This is what I understood what's going on here when talking with @lukmccall.

Reply:
I think that should be `expo-modules-core`, not `expo-modules-code`...

As I recall, the problem this is solving is due to some changes in Android libraries between different versions of React Native.

Reply:
@lukmccall - is there a linear task to follow up on this? also - thanks for helping @dsokal figure it out :)

Reply:
I think @lukmccall this was already fixed in https://github.com/expo/expo/pull/18607 but this is probably still not released?

Reply:
That's true. It was fixed, but not released.